server/fleet/orbit.go

```diff
@@ -50,3 +50,32 @@ type OrbitHostInfo struct {
 	// Platform is the device's platform as defined by osquery.
 	Platform string
 }
+
+// ExtensionInfo holds the data of a osquery extension to apply to an Orbit client.
+type ExtensionInfo struct {
+	// Platform is one of "windows", "linux" or "macos".
+	Platform string `json:"platform"`
+	// Channel is the select TUF channel to listen for updates.
+	Channel string `json:"channel"`
+}
+
+// Extensions holds a set of extensions to apply to an Orbit client.
+// The key of the map is the extension name (as defined on the TUF server).
+type Extensions map[string]ExtensionInfo
+
+// FilterByHostPlatform filters out extensions that are not targeted for hostPlatform.
+// It supports host platforms reported by osquery and by Go's runtime.GOOS.
+func (es *Extensions) FilterByHostPlatform(hostPlatform string) {
```

Comment:
feels like this method could use a couple tests

Reply:
Good idea. 3712d081dabac97125f1fffd7dd0c1b7bb25122c.